### ISSUE TITLE
Make it possible to configure the rspec command used by the build system

### ISSUE THREAD

**Issue by valo** [SublimeText/RSpec#50]
I am trying to use the build system on a rails 4.1 project. I am using RVM and the build system is failing with

```
[Errno 2] No such file or directory: 'rspec'
....
[path: /usr/bin:/bin:/usr/sbin:/sbin]
[Finished]
```

It will be nice to be able to override the cmd in the build system to use `./bin/rspec` instead of just rspec.

**Comment by jamonholmgren** [comment]
I'm having trouble getting this to work, too. I use [chruby](https://github.com/postmodern/chruby) by @postmodern and have tried many different combinations, with no luck.

Latest try:

```json
{
  "env": {
    "PATH": "/usr/local/opt/chruby/bin:${PATH}"
  },
  "cmd": ["chruby-exec", "ruby", "2.2.0", "--", "bundle", "exec", "rspec", "-I ${file_path}", "$file"],
  "file_regex": "# ([A-Za-z:0-9_./ ]+rb):([0-9]+)",
  "working_dir": "${project_path:${folder:${file_path}}}",
  "selector": "source.ruby.rspec"
}
```

Results in:

```
zsh:1: command not found: chruby
[Finished in 0.1s with exit code 127]
[cmd: ['chruby-exec', 'ruby', '2.2.0', '--', 'bundle', 'exec', 'rspec', '-I /Users/jh/Code/Rails/project/spec/controllers', '/Users/jh/Code/Rails/project/spec/controllers/products_controller_spec.rb']]
[dir: /Users/jh/Code/Rails/project]
[path: /usr/local/opt/chruby/bin:${PATH}]
```

Any ideas? Wish I could just spin up my normal shell environment easily and run `bundle exec rspec <file>`.

**Comment by princemaple** [comment]
Might have to go with a python command, but will lose many nice things the
build system has...

**Comment by jamonholmgren** [comment]
I figured out a way to do this in chruby using the RubyTest package instead.

For those of you interested:

**Keyboard Shortcuts**

```json
[
  { "keys": ["alt+shift+r"], "command": "run_all_ruby_test",
      "context": [ { "key": "selector", "operator": "equal",
         "operand": "source.ruby, source.rspec, text.gherkin.feature"
                } ]
  }, // test file

  { "keys": ["super+shift+t"], "command": "reopen_last_file" }, // un-hijack this key combo
]
```

**RubyTest.sublime-settings**

```json
{
  "run_rspec_command": "/Users/jh/rspec_env.sh bundle exec spring rspec {relative_path}",
  "run_single_rspec_command": "/Users/jh/rspec_env.sh bundle exec spring rspec {relative_path}:{line_number}",
  "save_on_run": true
}
```

Make sure to change the path to match your own user.

**~/rspec_env.sh**

This file loads the environment. I'm using zsh.

```
#!/usr/bin/env zsh
source ./.zshrc
$@
```

I could be doing this in a boneheaded way, but it's working for me, so hopefully it'll help someone else.

**Comment by princemaple** [comment]
That's basically what I meant by having to go the python command way. Might
try and make a cleaner implementation / interface.

**Comment by jamonholmgren** [comment]
Cool, thanks @princemaple. I like this package better than the RubyTest one, so definitely update here if there's a good way to do it in RSpec ST package.

**Comment by princemaple** [comment]
Update: I'm interested to see what new build system Jon is going to bring us.

**Comment by princemaple** [comment]
@jamonholmgren and @valo , just FYI, [rbenv](https://github.com/sstephenson/rbenv) could make your life much easier.

**Comment by jamonholmgren** [comment]
@princemaple Thanks for the advice. I'm pretty happy with chruby.

**Comment by ristovskiv** [comment]
I'm having the similar problem, but everything worked great until I've installed LiveReload package. Then suddenly after using the build command "Ctrl + B" it gave me this error. I've tried everything (unistalled LiveReload first of course) and can't seem to make it work again like it used to. Any help? I'm using RVM as well.

**Comment by princemaple** [comment]
With the new build system that ST3 rolled out recently, I think we can solve this problem by having multiple build files that target the same file type. So everyone can use whatever suits them the best.

@ristovskiv can you tell me what you would have to run manually to get spec running?

**Comment by ristovskiv** [comment]
Sorry if I didn't understand you properly, you mean what I need to run on the CLI so that the specs would run?

**Comment by princemaple** [comment]
Yes, please

On Thu, Mar 26, 2015 at 10:50 PM, Vlatko Ristovski <notifications@github.com
> wrote:

> Sorry if I didn't understand you properly, you mean what I need to run on
> the CLI so that the specs would run?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/SublimeText/RSpec/issues/50#issuecomment-86479915>.
>



-- 
Regards,

Po Chen

**Comment by ristovskiv** [comment]
I'm using just "rspec spec/" or just "rspec"  for all specs

**Comment by princemaple** [comment]
So no special commands for rvm? sorry I use rbenv, am not very familiar how
things are done with rvm.
rbenv doesn't need any special commands (too), but I've seen people saying
you have to run special commands for rvm.

On Thu, Mar 26, 2015 at 11:53 PM, Vlatko Ristovski <notifications@github.com
> wrote:

> I'm using just "rspec spec/" or just "rspec" for all specs
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/SublimeText/RSpec/issues/50#issuecomment-86501335>.
>



-- 
Regards,

Po Chen

**Comment by ristovskiv** [comment]
Nope, at least I've never used a special command to run the specs nor the tests. Anyway one funny thing. I didn't know that there was a new version of ST3 released today. So I've upgraded ST3 to Build 3083 and now everything works fine... I don't have a simple explanation how this happened. I've wasted a lot of time on this but I'm a really stubborn guy, but the problem is here that I've wasted your time as well. So my apologies for that . If something goes wrong I'll update here.

**Comment by princemaple** [comment]
Thanks a lot for the update. I think I will spend some time to update the
build for rspec package anyway, to take advantage of the new
functionalities of ST3.

**Comment by ristovskiv** [comment]
+1 for that. thanks!

**Comment by dylandrop** [comment]
Any update on this? I'm aiming to leverage [spring](https://github.com/rails/spring) to run my tests -- at the end of the day basically just means prepending `spring` to my tests. Any suggested workarounds would be much appreciated.

**Comment by princemaple** [comment]
@dylandrop some time ago I added https://github.com/SublimeText/RSpec/blob/master/RSpec%20-%20Bundle%20Exec.sublime-build to utilize ST3's new build system. I can simply add another one for `spring rspec $file`, if that is what you want. With spring, I recommend using [guard](https://github.com/guard/guard) to continuously watch your files and rebuild as soon as anything changes, it's fast with spring in play.

So yeah, send a PR if you want, or just tell me to add another build file for spring rspec.

**Comment by valo** [comment]
Yeah... I moved along with using `guard`. Never really managed to get this working and gave up after a while. The combination of RVM, complex env setup and sublime makes things complicated...

**Comment by dylandrop** [comment]
@princemaple Would that just involve copy/pasting the file you pointed to, and replacing the `"bundle", "exec"` with `"spring"`? I'd be happy to set up a PR to add that file, although I'm not sure how I'd actually then get Sublime to use this new command.

Not a big fan of guard since it always triggers testing the whole file every time I hit save, but I appreciate the advice.
